src/Management/src/Endpoint/Loggers/LoggersEndpointHandler.cs

```diff
@@ -23,37 +23,43 @@ internal sealed class LoggersEndpoint : ILoggersEndpoint
         LoggerLevels.MapLogLevel(LogLevel.Trace)
     };
 
-    private readonly ILogger<LoggersEndpoint> _logger;
+    private readonly ILogger<LoggersEndpointHandler> _logger;
     private readonly IOptionsMonitor<LoggersEndpointOptions> _options;
     private readonly IDynamicLoggerProvider _dynamicLoggerProvider;
 
-    public IEndpointOptions Options => _options.CurrentValue;
+    public HttpMiddlewareOptions Options => _options.CurrentValue;
 
-    public LoggersEndpoint(IOptionsMonitor<LoggersEndpointOptions> options, ILoggerFactory loggerFactory, IDynamicLoggerProvider dynamicLoggerProvider)
+    public LoggersEndpointHandler(IOptionsMonitor<LoggersEndpointOptions> options, ILoggerFactory loggerFactory, IDynamicLoggerProvider dynamicLoggerProvider)
     {
         ArgumentGuard.NotNull(options);
         ArgumentGuard.NotNull(loggerFactory);
         ArgumentGuard.NotNull(dynamicLoggerProvider);
 
         _options = options;
         _dynamicLoggerProvider = dynamicLoggerProvider;
-        _logger = loggerFactory.CreateLogger<LoggersEndpoint>();
+        _logger = loggerFactory.CreateLogger<LoggersEndpointHandler>();
     }
-
-    public Task<Dictionary<string, object>> InvokeAsync(LoggersChangeRequest request, CancellationToken cancellationToken)
+    public LoggersEndpointHandler(IOptionsMonitor<LoggersEndpointOptions> options, ILoggerFactory loggerFactory)
```

Comment:
Yes sometimes the `IDynamicLoggerProvider` is not available in the service container.

Reply:
It looks like we have code to handle it being null at request time, but would it be better if we could handle that concern in the constructor since this endpoint is (I think?) useless without it?

Reply:
good point, I was thinking of the IDynamicMessageProcessor.